src/credentials/attestation_verifier/JSONFileRevocationSet.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "JSONFileRevocationSet.h"
+
+#include <lib/support/Base64.h>
+#include <lib/support/BytesToHex.h>
+#include <lib/support/ScopedBuffer.h>
+
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+
+namespace chip {
+namespace Credentials {
+
+JSONFileRevocationSet::JSONFileRevocationSet(const char * revocationSetPath)
+{
+    VerifyOrReturn(revocationSetPath != nullptr);
+
+    if (revocationSetPath != nullptr)
+    {
+        LoadJSONRevocationSet(revocationSetPath, mRevocationSet);
+        VerifyOrReturn(revocationSetCount());
+    }
+
+    mIsInitialized = true;
+}
+
+void LoadJSONRevocationSet(const char * revocationSetPath, Json::Value & revocationSet)
+{
+    VerifyOrReturn(revocationSetPath != nullptr);
+
+    std::ifstream ifs(revocationSetPath);
+    Json::Reader reader;
+    reader.parse(ifs, revocationSet);
+}
+
+AttestationVerificationResult JSONFileRevocationSet::IsCertificateRevoked(bool isPaa, Crypto::AttestationCertVidPid vidPidUnderTest,
+                                                                          ByteSpan issuer, ByteSpan authorityKeyId,
+                                                                          ByteSpan serialNumber) const
+{
+    for (size_t revocation_set_idx = 0; revocation_set_idx < mRevocationSet.size(); ++revocation_set_idx)
+    {
+        Json::Value type = mRevocationSet[static_cast<int>(revocation_set_idx)]["type"];
+
+        // 1.
+        if (type.asString().compare("revocation_set") == 0)
+        {
+            Json::Value jsonCrlIssuerSubjectKeyId = mRevocationSet[static_cast<int>(revocation_set_idx)]["issuer_subject_key_id"];
+            Json::Value jsonCrlIssuerName         = mRevocationSet[static_cast<int>(revocation_set_idx)]["issuer_name"];
+            Json::Value jsonCrlRevokedSerialNumbers =
+                mRevocationSet[static_cast<int>(revocation_set_idx)]["revoked_serial_numbers"];
+
+            uint8_t crlIssuerSubjectKeyIdBuf[Crypto::kAuthorityKeyIdentifierLength] = { 0 };
+            ByteSpan crlIssuerSubjectKeyId(crlIssuerSubjectKeyIdBuf);
```

Comment:
Where does the size of this span get adjusted to the actual number of bytes HexToBytes produces?